CMakeLists.txt

```diff
@@ -1,11 +1,14 @@
-cmake_minimum_required(VERSION 3.21..3.25)
+cmake_minimum_required(VERSION 3.16...3.29)
```

Comment:
Why do we need to specify the maximum version here?

Reply:
It's so that CI can take advantage of new CMake policies. I don't remember if there was a particular one that was important, but in any case it doesn't really hurt anybody.